backend-db/Rhyolite/Backend/DB/PsqlSimple.hs

```diff
@@ -190,57 +190,33 @@ instance (FromField (IdData a)) => FromField (Id a) where
 instance (ToField (IdData a)) => ToField (Id a) where
   toField (Id x) = toField x
 
+defaultQQ :: String -> QuasiQuoter
+defaultQQ name = QuasiQuoter
+  { quotePat = error $ name <> ": quasiquoter used in pattern context"
+  , quoteType = error $ name <> ": quasiquoter used in type context"
+  , quoteExp = error $ name <> ": quasiquoter used in expression context"
+  , quoteDec = error $ name <> ": quasiquoter used in declaration context"
+  }
 
 -- | This quasiquoter is the obvious combination of 'sqlQ' and 'query'.
 queryQ :: QuasiQuoter
-queryQ = QuasiQuoter
-  { quotePat  = error "queryQ: quasiquoter used in pattern context"
-  , quoteType = error "queryQ: quasiquoter used in type context"
-  , quoteExp  = \s -> appE [| uncurry query |] (sqlQExp s)
-  , quoteDec  = error "queryQ: quasiquoter used in declaration context"
-  }
+queryQ = (defaultQQ "queryQ") { quoteExp = appE [| uncurry query |] . sqlQExp }
 
 -- | This quasiquoter is the obvious combination of 'sqlQ' and 'execute'.
 executeQ :: QuasiQuoter
-executeQ = QuasiQuoter
-  { quotePat  = error "executeQ: quasiquoter used in pattern context"
-  , quoteType = error "executeQ: quasiquoter used in type context"
-  , quoteExp  = \s -> appE [| uncurry execute |] (sqlQExp s)
-  , quoteDec  = error "executeQ: quasiquoter used in declaration context"
-  }
+executeQ = (defaultQQ "executeQ") { quoteExp = appE [| uncurry execute |] . sqlQExp }
 
 executeQ_ :: QuasiQuoter
-executeQ_ = QuasiQuoter
-  { quotePat  = error "executeQ: quasiquoter used in pattern context"
-  , quoteType = error "executeQ: quasiquoter used in type context"
-  , quoteExp  = \s -> appE [| uncurry execute_ |] (sqlQExp s)
-  , quoteDec  = error "executeQ: quasiquoter used in declaration context"
-  }
-
+executeQ_ = (defaultQQ "executeQ_") { quoteExp = appE [| uncurry execute_ |] . sqlQExp }
 
 traceQueryQ :: QuasiQuoter
-traceQueryQ = QuasiQuoter
-  { quotePat  = error "traceQueryQ: quasiquoter used in pattern context"
-  , quoteType = error "traceQueryQ: quasiquoter used in type context"
-  , quoteExp  = \s -> appE [| uncurry traceQuery |] (sqlQExp s)
-  , quoteDec  = error "traceQueryQ: quasiquoter used in declaration context"
-  }
+traceQueryQ = (defaultQQ "traceQueryQ") { quoteExp = appE [| uncurry traceQuery |] . sqlQExp }
 
 traceExecuteQ :: QuasiQuoter
-traceExecuteQ = QuasiQuoter
-  { quotePat  = error "traceQueryQ: quasiquoter used in pattern context"
-  , quoteType = error "traceQueryQ: quasiquoter used in type context"
-  , quoteExp  = \s -> appE [| uncurry traceExecute |] (sqlQExp s)
-  , quoteDec  = error "traceQueryQ: quasiquoter used in declaration context"
-  }
+traceExecuteQ = (defaultQQ "traceExecuteQ") { quoteExp = appE [| uncurry traceExecute |] . sqlQExp }
 
 traceExecuteQ_ :: QuasiQuoter
```

Comment:
Heh, `traceExecuteQ` and `traceExecuteQ_` actually had wrong error messages

Reply:
So technically this is a semantic change. You know...for those people who are `catch`ing this and making choices based on the string of the message. :thinking: :wink: